stonadsstatistikk_ef/src/main/kotlin/no/nav/familie/eksterne/kontrakter/ef/Vedtak.kt

```diff
@@ -0,0 +1,107 @@
+package no.nav.familie.eksterne.kontrakter.ef
+
+import java.time.LocalDate
+import java.time.ZonedDateTime
+
+data class Vedtak(
+        val sakId: String,
+        val behandlingId: String,
+        val relatertBehandlingId: String,
+        val kode6eller7: Boolean,
+        val tidspunktVedtak: ZonedDateTime,
+        val vilkårsvurderinger: List<Vilkårsvurdering>,
+        val person: Person,
+        val barn: List<Person>,
+        val behandlingType: BehandlingType,
+        val behandlingÅrsak: BehandlingÅrsak,
+        val behandlingResultat: BehandlingResultat,
+        val vedtakResultat: VedtakResultat? = null,
+        val utbetalinger: List<Utbetaling>,
+        val inntekt: List<Inntekt>,
+        val aktivitetskrav: Aktivitetskrav,
+        val funksjonellId: String? = null,
+)
+
+enum class BehandlingType {
+    FØRSTEGANGSBEHANDLING,
+    REVURDERING,  // Inkluderer opphør
+    KLAGE,
+    MIGRERING_FRA_INFOTRYGD,
+    TILBAKEFØRING_TIL_INFOTRYGD
+}
+
```

Comment:
Trenger litt input på disse enum'ene, type:

- BehandlingType
- BehandlingÅrsak
- BehandlingResultat
- Vedtak(Resultat)

Hadde vært fint om disse representerte ulike "ting". Nå er det en del overlapp/utydelighet

Reply:
@tnarland @henninghaakonsen kan kanskje svare på dette bedre enn meg

Reply:
Vi trenger alle, men i ba-sak er nok vedtakResultat==behandlingResultat.

BehandlingType definerer hva slags behandling det er: førstegangsbehandling, migrering etc
BehandlingÅrsak definerer hvorfor behandlingen ble opprettet.
BehandlingResultat definerer resultatet for inneværende behandling (altså endringene på den ene behandlingen).

Reply:
Gir det mening, @henninghaakonsen, at det skilles mellom `BehandlingResultat` og `Vedtak`? Nå er tolkningen at bare `BehandlingResultat.FULLFØRT` fører til at `Vedtak` har en verdi.

Reply:
Hmm, det skjønte jeg ikke helt. Du er forsåvidt avhengig av et behandlingsresultat for å lage et vedtak i ba-sak, men det er jo bare fordi vi har modellert det slik. 

Uansett, det som er interessant for DVH er vel hva behandlingen førte til (behandlingsresultat) og når det ble iverksatt (vedtaksdato). Annet er vel ikke så relevant?